### PR TITLE
Upgrade to angular 6

### PR DESCRIPTION
**Pull request by rvalenciano**
## Summary

Updating package to use angular 6

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,45 +36,59 @@
   ],
   "private": false,
   "dependencies": {
-    "bootstrap": "4.0.0-beta.2"
+    "@angular/animations": "^6.0.6",
```
Comment (yp2) [inline package.json]: @rvalenciano should in dependencies be only bootstrap - rest of it should go to peerDependencies?

File: src/environments/environment.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // The build system defaults to the dev environment which uses `environment.ts`, but if you do
 // `ng build --env=prod` then `environment.prod.ts` will be used instead.
 // The list of which env maps to which file can be found in `.angular-cli.json`.
+import 'zone.js/dist/zone-error';
```
Comment (yp2) [inline src/environments/environment.ts]: import in environment? It is required?

**Comment by yp2** [comment]
@rvalenciano build:docs script in package.json has not updated param: -op should be change to --output-path. This build is crashing. We need this to generate demo site.

File: package.json (hunk 1 above)
Comment (rvalenciano) [reply package.json]: Correct me if I'm not wrong (not an angular expert per se), but if I remove these dependencies I got a bunch of lines like this:

```shell
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/animations@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/common@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/compiler@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/core@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/forms@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/http@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/platform-browser@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/platform-browser-dynamic@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-cron-jobs@6.4.9 requires a peer of @angular/router@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/compiler-cli@6.0.6 requires a peer of @angular/compiler@6.0.6 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.0.0-beta.2 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.0.0-beta.2 requires a peer of popper.js@^1.12.3 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.4.0 requires a peer of @angular/compiler@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.4.0 requires a peer of @angular/core@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN karma-jasmine-html-reporter@1.1.0 requires a peer of jasmine@>=3 but none is installed. You must install peer dependencies yourself.
npm WARN rxjs-marbles@2.4.1 requires a peer of rxjs@^5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN The package bootstrap is included as both a dev and production dependency.
```
And if I add them in dependencies this messages stop. Actually trying to run for example `ng test --source-map=false` breaks with error 

```shell
Could not find API compiler-cli, function VERSION
Error: Could not find API compiler-cli, function VERSION
``` 

If I add these dependencies then everything it's fix. I tried to google this issue but got no luck. Maybe I'm missing something?

File: src/environments/environment.ts (hunk 2 above)
Comment (rvalenciano) [reply src/environments/environment.ts]: This is just to have detailed errors from zone.js. If you want me I can remove it.

File: package.json (hunk 1 above)
Comment (yp2) [reply package.json]: Ok. I think we may leave it. @rvalenciano maybe we don't need peerDependencies?

File: src/environments/environment.ts (hunk 2 above)
Comment (yp2) [reply src/environments/environment.ts]: ok

File: package.json (hunk 1 above)
Comment (rvalenciano) [reply package.json]: According to:
https://nodejs.org/en/blog/npm/peer-dependencies/

Peer Dependencies should be used in a scenario like following:
```shell
├── request@2.12.0
└─┬ some-other-library@1.2.3
  └── request@1.9.9
```

So in this case I think that yes, we can remove peer dependencies.

Comment (yp2) [reply package.json]: I will merge this PR. @rvalenciano i will remove it on master after merge.

**Pull request merged**